src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -2575,6 +2575,9 @@ private void initDataPos() throws IOException {
                 if (readNBytesAt(buf, 0, buf.length, pos) != LOCHDR) {
                     throw new ZipException("invalid loc " + pos + " for entry reading");
                 }
+                if (LOCSIG(buf) != LOCSIG) {
+                    throw new ZipException("invalid loc header (bad signature)");
```

Comment:
Please change loc -> LOC

Reply:
Fixed, including the ZipException for the preceding read check.